src/Plugin.php

```diff
@@ -63,6 +63,16 @@ public static function wp_enqueue_scripts() {
     wp_enqueue_script('gallerya-script-custom', static::getBaseUrl() . '/dist/scripts/script.min.js', ['gallerya-script-libs'], FALSE, TRUE);
   }
 
+  public static function post_gallery_init() {
+    $url = home_url(add_query_arg(array()));
+    $id = url_to_postid($url);
+    $post_type = '';
+    if (get_post_type($id) === $post_type || get_post_type() == '') {
+      add_filter('post_gallery', __CLASS__ . '::post_gallery', 10, 2);
+    }
+    return apply_filters('post_gallery_init', $post_type);
```

Comment:
In Wordpress you have to distinguish between [actions](https://codex.wordpress.org/Glossary#Action) and [filters](https://codex.wordpress.org/Glossary#Filter). Actions are used to inject custom code into an existing function/output like:

```
// my-template.php:
<div>
 My output

 <?php do_action('my-action'); ?>

 The other output
</div>
```

```php
// functions.php:
function my_action() {
  echo 'My action output!';
}
add_action('my-action', 'my_action');
```
This will result in:
```
// source code:
<div>
 My output

 My action output!

 The other output
</div>
```

While filters are used to actually filter a given value like:

```php
function my_function() {
    $post_type = apply_filters('my-custom-filter', 'post');
}

function my_custom_filter($post_type) {
  $post_type = 'page';
  return $post_type;
}
add_filter('my-custom-filter', 'my_custom_filter');
```
This will change the post type from post to page.

Actions echo code, while functions must return the parameter. It is also possible to give arguments by reference like the `pre_get_posts` action, where you can manipulate the actual `$query` object.